azurerm/internal/services/springcloud/spring_cloud_java_deployment_resource.go

```diff
@@ -0,0 +1,248 @@
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SpringCloudDeploymentName,
+			},
+
+			"spring_cloud_app_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SpringCloudAppID,
+			},
+
+			"cpu": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntBetween(1, 4),
+			},
+
+			"environment_variables": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"instance_count": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntBetween(1, 500),
+			},
+
+			"jvm_options": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"memory_in_gb": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntBetween(1, 8),
+			},
+
+			"runtime_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(appplatform.Java8),
+					string(appplatform.Java11),
+				}, false),
+				Default: string(appplatform.Java8),
+			},
+		},
+	}
+}
+
+func resourceSpringCloudJavaDeploymentCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).AppPlatform.DeploymentsClient
+	servicesClient := meta.(*clients.Client).AppPlatform.ServicesClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	appId, err := parse.SpringCloudAppID(d.Get("spring_cloud_app_id").(string))
+	if err != nil {
+		return err
+	}
+
+	id := parse.NewSpringCloudDeploymentID(subscriptionId, appId.ResourceGroup, appId.SpringName, appId.AppName, d.Get("name").(string))
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, id.ResourceGroup, id.SpringName, id.AppName, id.DeploymentName)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for presence of existing Spring Cloud Deployment %q (Spring Cloud Service %q / App %q / Resource Group %q): %+v", id.DeploymentName, id.SpringName, id.AppName, id.ResourceGroup, err)
+			}
+		}
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_spring_cloud_java_deployment", id.ID())
+		}
+	}
+
+	service, err := servicesClient.Get(ctx, appId.ResourceGroup, appId.SpringName)
+	if err != nil {
+		return fmt.Errorf("checking for presence of existing Spring Cloud Service %q (Resource Group %q): %+v", appId.SpringName, appId.ResourceGroup, err)
+	}
+
+	skuName := "S0"
+	skuTier := "Standard"
+	if sku := service.Sku; sku != nil {
+		if sku.Name != nil {
+			skuName = *sku.Name
+			skuTier = *sku.Tier
+		}
+	}
```

Comment:
Is there a valid case where we'd get an empty response back for these values?  Feels like we should error if we don't have concrete values, rather than assume them?

Reply:
there should be no such case, I am just doing Defensive programming. I have changed it to throw error.